src/main.rs

```diff
@@ -40,6 +45,20 @@ fn main() {
     // we can have an infinite loop that will read these
     // for files.
     loop {
+
+        if(args.len() > 1) {
+          if(args[1] == "-t") {
+            let start2 = SystemTime::now();
+            let since_the_epoch2 = start2
+                .duration_since(UNIX_EPOCH)
+                .expect("Time went backwards"); 
+            if(since_the_epoch2.as_millis() - since_the_epoch.as_millis() > args[2].parse::<u128>().unwrap()) {
+              break;
```

Comment:
This use of `since_the_epoch2` and `start2` is really a bad practice - see if you can clean this up. (if not, that's completely fine too).